### PR TITLE
removeSelected prop (round 2), for optionally keeping selected values in dropdown

### PR DESCRIPTION
**Pull request by banderson**
I forked what @dherran did with #882 to resolve conflicts with master and to implement PR feedback from @agirton, so defer to #882 and the original issue for more details: #803 

This is in a holding state until we decide what to do here. But this should work as expected, and all tests are passing.

### CODE REVIEW

**Comment by agirton** [comment]
Hi @banderson thank you for this. Since this adds a new prop this will need unit tests and documentation. Can you add these?

**Comment by banderson** [comment]
@agirton I just updated this to include:

 - Documentation in README to describe this feature: 3d88a771ea83ad70636cebeeea21869f4e659611
 - Test coverage for selecting (878b88041ba539b730ade50b390cc9fc2023f5e5) and deselecting (3779f382ee0e4fe557ee1bf9cfd335bdfd7c2593) values

The original author (@dherran) already added docs to the example site.

I tried to follow the testing style that was there, let me know if there's anything else I can do to help get this in!

File: README.md [hunk 1]
```diff
@@ -104,7 +104,7 @@ The built-in Options renderer also support custom classNames, just add a `classN
 
 You can enable multi-value selection by setting `multi={true}`. In this mode:
 
-* Selected options will be removed from the dropdown menu
+* Selected options will be removed from the dropdown menu by default. If you want them to remain in the list, set `removeSelected={true}`
```
Comment (gwyneplaine) [inline README.md]: @banderson  just a minor nit: should this  be "If you want them to remain in the list, set `removeSelected={false}`"?

**Comment by ansgarm** [comment]
Is there anything currently blocking this (besides the true -> false change in the docs)?

We could really need this aswell and I'd be happy to help with any remaining steps so this PR can be merged :)

**Comment by ansgarm** [comment]
Two test cases were broken btw, fixed them here:
https://github.com/elbstack/react-select/commit/4c744b5afc6e21964b6b53324a5e7c87fd9ac9c9

**Comment by JedWatson** [comment]
Thanks everyone! This is good to land 👍

**Pull request merged**